core/src/main/scala/org/apache/spark/storage/DiskBlockManager.scala

```diff
@@ -91,7 +90,12 @@ private[spark] class DiskBlockManager(blockManager: BlockManager, conf: SparkCon
   /** List all the files currently stored on disk by the disk manager. */
   def getAllFiles(): Seq[File] = {
     // Get all the files inside the array of array of directories
-    subDirs.flatten.filter(_ != null).flatMap { dir =>
+    subDirs.flatMap { dir =>
```

Comment:
OK, imagine this code:

```
var a: String = null;
...
def foo() = {
   var b = "foo"
   b = b + "bar"
   a = b;
 }
```

No matter how a thread observes `a`, it can't see `a` have value `"foo"`. Right? It can observe `null` or `"foobar"`. That's the situation here.